test/node-rpc-test.js

```diff
@@ -5,59 +5,512 @@
+    return {
+      apiKey: nodeOptions.apiKey,
+      timeout: TIMEOUT,
+      port: port
+    };
+  };
 
-describe('RPC', function() {
-  this.timeout(15000);
+  const fullnodeAddr = `127.0.0.1:${ports.full.p2p}`;
+  const fullnode = new FullNode({
+    listen: true,
+    bip37: true,
+    ...nodeOptions,
+    ...getNodePorts(ports.full)
+  });
+
+  const spvnode = new SPVNode({
+    only: fullnodeAddr,
+    ...nodeOptions,
+    ...getNodePorts(ports.spv)
+  });
+
+  const prunednode = new FullNode({
+    prune: true,
+    only: fullnodeAddr,
+    ...nodeOptions,
+    ...getNodePorts(ports.pruned)
+  });
+  const fullindexed = new FullNode({
+    only: fullnodeAddr,
+    indexTX: true,
+    ...nodeOptions,
+    ...getNodePorts(ports.fullindexed)
+  });
+
+  const nclient = new NodeClient({...getClientOpts(ports.full.node)});
+  const wclient = new WalletClient({...getClientOpts(ports.full.wallet)});
+
+  const spvnclient = new NodeClient({...getClientOpts(ports.spv.node)});
+  const prunednclient = new NodeClient({...getClientOpts(ports.pruned.node)});
+  const indexednclient = new NodeClient({
+    ...getClientOpts(ports.fullindexed.node)
+  });
+
+  const MINER_WALLET = 'test';
 
   before(async () => {
-    await node.open();
+    await fullnode.open();
+    await fullnode.connect();
+
+    await prunednode.open();
+    await prunednode.connect();
+    prunednode.startSync();
+
+    await spvnode.open();
+    await spvnode.connect();
+    spvnode.startSync();
+
+    await fullindexed.open();
+    await fullindexed.connect();
+    fullindexed.startSync();
+
+    // create and select miner wallet.
+    await wclient.createWallet(MINER_WALLET);
+    await wclient.execute('selectwallet', [MINER_WALLET]);
+
+    // setup miner wallet
+    const address = await wclient.execute('getnewaddress');
+    const {wdb} = fullnode.require('walletdb');
+
+    // generate coins to our miner.
+    await nclient.execute('generatetoaddress', [101, address]);
+
+    // wait for wallet to connect blocks.
+    await forValue(wdb, 'height', 101);
+    await forValue(fullindexed.txindex, 'height', 101);
+    await forValue(prunednode.chain, 'height', 101);
+    await forValue(spvnode.chain, 'height', 101);
   });
 
   after(async () => {
-    await node.close();
+    fullindexed.stopSync();
+    spvnode.stopSync();
+    prunednode.stopSync();
+
+    await spvnode.disconnect();
+    await prunednode.disconnect();
+    await fullindexed.disconnect();
+    await fullnode.disconnect();
+
+    await fullnode.close();
+    await prunednode.close();
+    await spvnode.close();
+    await fullindexed.close();
   });
 
-  it('should rpc help', async () => {
-    assert(await nclient.execute('help', []));
+  describe('help', function() {
+    it('should list all methods', async () => {
+      const response = await nclient.execute('help', []);
+      const lines = response.split('\n');
+
+      assert(response);
+      assert(lines.length);
```

Comment:
Is it crazy to check this value against `Object.keys(fullnode.rpc.calls).length + 1` ?